jina/types/list.py

```diff
@@ -0,0 +1,64 @@
+from collections.abc import MutableSequence
+from typing import Union
+
+from google.protobuf import struct_pb2
+
+from .mixin import ProtoTypeMixin
+
+
+class ListView(ProtoTypeMixin, MutableSequence):
+    """Create a Python mutable sequence view of Protobuf ListValue struct.
+
+    This can be used in all Jina types where a protobuf.ListValue is returned.
+
+    Used inside `StructView` when the inner element is a `ListValue`
+    """
+
+    def __init__(self, list: struct_pb2.ListValue):
+        """Create a Python list view of Protobuf ListValue.
+
+        :param list: the protobuf ListValue object
+        """
+        self._pb_body = list
+
+    def insert(self, index: int, object) -> None:
+        """
+        Insert any proto object into the list
+
+        :param index: Position of the insertion.
+        :param object: The object that needs to be inserted.
+        """
+        self._pb_body.insert(index, object)
+
+    def __getitem__(self, i: Union[int, slice]):
+        if i < len(self):
```

Comment:
Raise an error when the index is out of range.
```suggestion
        if i >= len(self):
            raise IndexError('list index out of range')
        ...
```

Reply:
Done!